### PR TITLE
CLN: de-kludge Block.quantile

### PR DESCRIPTION
**Pull request by jbrockmendel**
Broken off from #27626 since I decided that is poorly scoped.

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -2564,6 +2540,19 @@ def equals(self, other):
             return False
         return (self.values.view("i8") == other.values.view("i8")).all()
 
+    def quantile(self, qs, interpolation="linear", axis=0):
+        naive = self.values.view("M8[ns]")
+
+        # kludge for 2D block with 1D values
```
Comment (gfyoung) [inline pandas/core/internals/blocks.py]: "kludge" - am I to assume that you're planning on cleaning this up or splitting more cleanly in the future?  One-liners like these don't seem "super-kludgy" IMO

Comment (jbrockmendel) [reply pandas/core/internals/blocks.py]: Yes, once we allow 2D EAs this entire method will be unnecessary.

**Review by TomAugspurger — APPROVED** [review]
LGTM. @jreback any thoughts? Otherwise I'd recommend self-merging in 24 hours @jbrockmendel.

**Comment by jreback** [comment]
@TomAugspurger i will look soonish. but I have comments on nearly all of these PRs so pls wait.

**Comment by jbrockmendel** [comment]
@jreback there's nothing urgent here, but you're gonna like the follow-up that comes after this and #27628.

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -1526,18 +1526,7 @@ def quantile(self, qs, interpolation="linear", axis=0):
...
-            # but `Block.get_values()` returns an ndarray of objects
-            # right now. We need an API for "values to do numeric-like ops on"
-            values = self.values.view("M8[ns]")
-
-            # TODO: NonConsolidatableMixin shape
-            # Usual shape inconsistencies for ExtensionBlocks
-            values = values[None, :]
-        else:
-            values = self.get_values()
+        values = self.get_values()
```
Comment (jreback) [inline pandas/core/internals/blocks.py]: grr I really dis-like strongly get_values()

**Review by jreback — APPROVED** [review]
comment for followup

**Pull request merged**